### PR TITLE
Travis: additional builds and Codecov

### PR DESCRIPTION
**Pull request by cbaker6**
Adds the following:

- [x] Builds: watchOS, tvOS (I'm not sure if any of the pre-builds are actually needed as `swift test` might cover all of them, but they don't take much time)
- [x] Adds codecov
- [x] Adds jazzy

### CODE REVIEW

**Comment by TomWFox** [comment]
I think the testing should be done in a separate 'target' rather than in SPM because that doesn't make much sense.

**Comment by cbaker6** [comment]
@TomWFox I'm not sure what you mean? You normally do `swift build` and `swift test` to make sure the your SPM package file works. 

It's possible to run the tests for specific devices on top of the "Build" phase in travis, but this might be redundant as `swift test` should cover if there's any issues.

File: Scripts/jazzy.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ver=`/usr/libexec/PlistBuddy -c "Print CFBundleShortVersionString" ParseSwift-iOS/Info.plist`
+jazzy \
+  --clean \
+  --author "Parse Community" \
+  --author_url http://parseplatform.org \
+  --github_url https://github.com/parse-community/Parse-Swift \
+  --root-url http://parseplatform.org/Parse-Swift/api/ \
```
Comment (cbaker6) [inline Scripts/jazzy.sh:7]: @TomWFox I just "made this up" based off of the other projects, not sure if you want this to be the link

**Comment by TomWFox** [comment]
> @TomWFox I'm not sure what you mean? You normally do `swift build` and `swift test` to make sure the your SPM package file works.

I'm not familiar with how SPM packages are normally tested etc - I just don't find it intuitive to look in the SPM build to see the test results - for the obj-c SDK the tests are split into different builds to the validation of the package manager configs. Let's see if anyone else chimes in with an opinion on this.

File: Scripts/jazzy.sh (hunk 1 above)
Comment (TomWFox) [reply Scripts/jazzy.sh:7]: I'm happy with that. The obj-c SDK currently published the jazzy docs on release (although it's broken rn!) - I'm thinking we could set this SDK to do it every commit - for now at least - what do you think?

**Comment by cbaker6** [comment]
I see, I think there will be some differences between this and the obj-c one because it currently can't use SPM. One example I have is my CertificateSigningRequest framework. The way I use keychain during the tests, allows them to pass for iOS, but the same keychain commands can't pass in OSX. To detect this in Xcode, I have to build and run the tests for both. When I run, `swift test`, it picks up and fails the test.

Another example is one of the [PRs](https://github.com/carekit-apple/CareKit/pull/459) I submitted to Apples CareKit framework. Basically, excluding a file needed to be added to build and test on watchOS. `swift build` and `swift test` was able to catch this.

I'm not sure of the limitations of this, but I can look into running the tests on each simulator which shouldn't be too hard.

File: Scripts/jazzy.sh (hunk 1 above)
Comment (cbaker6) [reply Scripts/jazzy.sh:7]: Sounds good to me!

File: .travis.yml [hunk 2]
```diff
@@ -25,11 +25,22 @@ jobs:
...
       script: carthage build --no-skip-current
+    - stage: Package Managers
+      script:
+        - gem install jazzy
+        - ./Scripts/jazzy.sh
+      deploy:
+        - provider: pages
+          skip_cleanup: true
+          token:
+            secure: ""
```
Comment (cbaker6) [inline .travis.yml:42]: @TomWFox can you add/setup the token?

**Comment by cbaker6** [comment]
@TomWFox I made the iOS build run the tests also which can be viewed [here](https://travis-ci.org/github/parse-community/Parse-Swift/jobs/713512685). SPM covers the macOS testcases at least. I didn't setup the watchOS and tvOS tests, though the builds are passing. I think there's way to setup the schemes not have to create a bunch of TestHost's (I'll look into this later)

File: .travis.yml (hunk 2 above)
Comment (TomWFox) [reply .travis.yml:43]: You haven't got maintainer edits turned on, here is the secure var - 

bW1tnO9sz1ehtyK4IbcjEUzDg7WvpMxd6wBz1BDxNi0IA4Qlv7KEd3+7gDL6xeNLue6OyVmCrLRoGpLQZmab21/FsX3lx87IV4xUJrg6wmruyUBtPWrTaHMTRHBsJFONsY0ADAn/MxXyVJyoK38gfnAUT6Q5WdfkTcn3jbB0sAqPjbd8JNJpfw4cAsxJjFL4Y4NDYIS8FWw/3uaEyX6JlNi6WOuomt3JKld91ioGH1gyAPoSDMufJPAm63jCGp/ichs4e/xLZ2XOziqgHZ6ujie9cTEf7cPk+xzzr1kuLFRMF9TPjO3Tf9MARK/vDoAMlR4a1h2q7wc6ok6FXGT37/01L7sX6JPzMBM8M5GJJpCaDIHj0PNG5QaN291kS7Vi7flLC8+ZkCddcPKh89m2uh7/FLNi3xh9RTazG0nq/Fztm3LlKzBRenF7aO9DCoOC2PFhg5vGuQk8N7SvK1A2kygjSJcj8gRqB3D0XnrcoNkXDiNjjgC01vhKOoHroTT0/Ds+KP0k6n9C/Hra0L5my7w0qnIkMffp6DW/VxMp3k+ZEJ4ZhVwCZ3pXJyon92tiqBweR0c0BjRpwyggJqIWxOeDJAM7v1Q6MoI1dO1sFQz7fDLh1VhcJpG4d7sg6J6mykotMDEd5re6VWwlpWf3ts7kYwldblq6SWr2Jk4yfL0=

Comment (cbaker6) [reply .travis.yml]: @TomWFox I'm thinking GitHub might have temporarily disabled "all maintainer edits" as I don't see it on the PR like how I use to on others and I usually leave that box checked

Comment (TomWFox) [reply .travis.yml]: Oh, maybe - anyway you can branch from the main repo now you've got write access

**Comment by cbaker6** [comment]
@TomWFox can you enable circleci?

**Comment by TomWFox** [comment]
I think the problem with the secure token was that I generated it for travis-ci.com but the project is on travis-ci.org. Give this one a go...

qgW5RfY0wfegdZ3qNZ/33XxwTRTol9GuX8g1nQkXGUm6gbrAXKjpWdg72jmBDGCZU4RZxzGJP/REtf+tCtJ8e1iEQIpJd/qcbFcb1r99if2ZZJB4IzNDsSafLhJF23lCJRoVRnNVgOxDfT6R1wywAWM7J9SogEvQwd2Ghw9uS98DAV6pm4zZlp+obOU1NYz4bvxkzKiunbOJeEJ450psvDOnZRMmAbunaopnG/tzYtA/NWGFCjARNmhD6N/3LeCegK1KLiOVF1J2Hob0MLom6NW5arvvFVYBTq2dEYNmgpO0A0Gabhr/S7OkXhNWkVBjsNfnRL3OmcPmTDpC9iLoES7zWfd6jHrrUv7wiLrmFoEdOaRrv0fELUaKsHITIjJDWZ7GD57fxHBmcu7pnajEUT0N0s8wXA0Cor5xYLiBrDJLdcV0TaREWpG6gB+XI7+YtkKryTySEv5Lnoq6XqGd4lM5P8IRfjyvLqUzdOjCpUeTcs44u5HVoXK5S+BfR8BXmoc1sz6Z0+gkJfB6twA4SBOlD2wwblwj7MvN9/2nFd8HHPTUuETvkHX15udAuS0R9a3Wu2ENQrWmA5xsdk2+2Jhqdnq3j+ZE8gyuXmwqLbP08q3qnqvOS+/lcy3A9/73Y1ad5XPHhuw6tNg6RCTI7x8mToIl6a8EU0AcVnp0EME=

**Comment by cbaker6** [comment]
@TomWFox the token still has an issue for some reason. 

I'll look into the how to set it up for circle also.

**Comment by cbaker6** [comment]
@TomWFox this is ready for review. I think we can work on the page deployment token issue directly by creating a branch.

This one is all travis and circle changes. The .h files and xml style files are autogenerated by Xcode and deal with me adding watchOS, tvOS builds to the project.

I suggest we merge this PR before the rest of them as this will see the codecov increases of the rest of the PRs along with CI adjustments

**Comment by cbaker6** [comment]
@TomWFox about what we talked about earlier in reference to `swift build` and `swift test` being cross-platform, here's a link to [AssetLib](https://github.com/brightdigit/AssetLib) being cross platform (look at their badges) and only using swift build/install to test. They use AssetLib and a really good article (I'm sure it's the same author), [Swift Package Continuous Integration Guide](https://learningswift.brightdigit.com/swift-package-continuous-integration-guide/). In AssetLib, they don't even have and xcodeproj, but instead tests if swiftpm can create the Xcode project successfully.

Basically, when you don't specify a platform, you are building testing for all devices with the caveat that you need to actually build/test on Linux. Parse-Swift currently doesn't work on Linux as the Foundation library we use currently doesn't have Keychain and URLSession support so the tests don't pass.

I specify the platforms in Package.swift so it can't be imported to Linux for the reason above. In the future, I'm sure we can remove the other build/tests, but they aren't hurting anything for now.
